### PR TITLE
Support for servant-0.16 WIP

### LINKED ISSUE #30: support servant-0.16
Thanks @snoyberg for reporting this.

servant-0.16 is out and there are several breaking changes for us to work around.

### PR DESCRIPTION
**Pull request by schell**
Fixes #30

### CODE REVIEW

**Comment by cdepillabout** [comment]
Thanks for working on this!  Once you get it working I'll happily merge it in and make a release!

**Comment by schell** [comment]
These changes compile and pass tests with `servant-0.16` but I'm still integrating it into a project to see if it works as expected. I've set the PR so you can review at the same time :)

File: stack.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # For more information, see: http://docs.haskellstack.org/en/stable/yaml_configuration.html
 
 # Specifies the GHC version and set of packages available (e.g., lts-3.5, nightly-2015-09-21, ghc-7.10.2)
-resolver: lts-11.4
+resolver: lts-13.16
```
Comment (cdepillabout) [inline stack.yaml]: Could you make this a recent version of `nightly`?

That way, you should be able to remove all the `servant-*` `extra-deps` below.

Comment (schell) [reply stack.yaml]: I'm trying out `nightly-2019-04-05` right now.

**Comment by schell** [comment]
@cdepillabout ready to go :)

**Comment by cdepillabout** [comment]
@schell I fixed up the tests so they are passing, merged, and made a release:

http://hackage.haskell.org/package/servant-checked-exceptions-core-2.1.0.0
http://hackage.haskell.org/package/servant-checked-exceptions-2.1.0.0

Thanks!

**Pull request merged**